### PR TITLE
New Libraries for 4.3.0!

### PR DESCRIPTION
**Pull request by korydraughn**
Contains required fixes for connection_pool library.

### CODE REVIEW

File: lib/filesystem/include/filesystem/path.hpp [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        path() = default;
+        path(const path& _p) = default;
+        path(path&& _p) noexcept = default;
+
+        path(const value_type* _source)
+            : value_{_source}
+        {
+        }
+
+        path(value_type* _source)
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply lib/filesystem/include/filesystem/path.hpp]: This is intended.  All cases experiencing this should be ignored as was.

File: lib/filesystem/include/filesystem/collection_entry.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#include "filesystem/object_status.hpp"
+
+#include <chrono>
+
+namespace irods::experimental::filesystem
+{
+    class collection_iterator;
+    class recursive_collection_iterator;
+
+    class collection_entry
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply lib/filesystem/include/filesystem/collection_entry.hpp]: The compiler will provide the special member functions.

File: lib/filesystem/include/filesystem/path.hpp [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        // Iterators
+
+        auto begin() const -> iterator;
+        auto end() const -> iterator;
+
+        auto rbegin() const -> reverse_iterator;
+        auto rend() const -> reverse_iterator;
+
+    private:
+        void append_seperator_if_needed(const path& _p);
```
Comment (trel) [inline lib/filesystem/include/filesystem/path.hpp]: separator

File: lib/filesystem/src/path.cpp [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+    {
+        return reverse_iterator{end()};
+    }
+
+    auto path::rend() const -> reverse_iterator
+    {
+        return reverse_iterator{begin()};
+    }
+
+    auto path::append_seperator_if_needed(const path& _p) -> void
```
Comment (trel) [inline lib/filesystem/src/path.cpp]: separator

File: server/core/include/rsLog.hpp [hunk 5]
```diff
@@ -8,34 +8,36 @@
...
-#define DEF_LOG_DIR "../../var/lib/irods/log"
-#define PROC_LOG_DIR_NAME "proc"
+#define DEF_LOG_DIR         "../../var/lib/irods/log"
+#define PROC_LOG_DIR_NAME   "proc"
+
 // =-=-=-=-=-=-=-
 // JMC - backport 4793
-#define DEF_LOGFILE_INT 5             /* default interval in days */
-#define LOGFILE_INT     "logfileInt"  /* interval in days for new log file */
+#define DEF_LOGFILE_INT     5                   /* default interval in days */
```
Comment (trel) [inline server/core/include/rsLog.hpp]: don't need these

**Pull request merged**